### PR TITLE
Omnibus PR (modelica datetime, REST API, worker scaling tests)

### PR DESCRIPTION
**Pull request by TShapinsky**
I needed to do this to make worker scale testing easier (being able to use the same inputs for both types of models).

### CODE REVIEW

File: alfalfa_worker/jobs/modelica/step_run.py [hunk 1]
```diff
@@ -8,18 +8,15 @@
 
 
 class StepRun(StepRunBase):
-    def __init__(self, run_id, realtime, timescale, external_clock, start_datetime="0", end_datetime=str(60 * 60 * 24 * 365)) -> None:
+    def __init__(self, run_id, realtime, timescale, external_clock, start_datetime: datetime, end_datetime) -> None:
```
Comment (nllong) [inline alfalfa_worker/jobs/modelica/step_run.py:11]: I like the explicit use of the inputs now.

**Review by nllong — DISMISSED** [review]
I think this is the right thing to do, but diehard modelica users will probably push back a bit because they are used to seconds from the beginning as the interface. I like this because it standardizes the two steps between OS and FMU.

File: alfalfa_web/server/resolvers.js [hunk 2]
```diff
@@ -243,8 +243,8 @@ async function sitesResolver(user, siteRef, context) {
           datetime: ""
         };
         const site_hash = await getHash(context.redis, site.siteRef);
-        if (site_hash && site_hash["sim_time"]) {
```
Comment (TShapinsky) [inline alfalfa_web/server/resolvers.js]: this was an important check, cause sometimes the site_hash is undefined. Without this check the sites view breaks

File: alfalfa_worker/lib/alfalfa_connections_manager.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import os
+
+import boto3
+from mongoengine import connect
+from redis import Redis
+
+from alfalfa_worker.lib.singleton import Singleton
+
+
+class AlafalfaConnectionsManager(metaclass=Singleton):
```
Comment (nllong) [inline alfalfa_worker/lib/alfalfa_connections_manager.py:10]: nice use of singleton

File: tests/jobs/openstudio/conftest.py [hunk 4]
```diff
@@ -1,14 +0,0 @@
-
-import os
-from pathlib import Path
-
-
-def pytest_generate_tests(metafunc):
```
Comment (nllong) [inline tests/jobs/openstudio/conftest.py:6]: very nice getting rid of this.

**Review by nllong — APPROVED** [review]
Looks great! This is definitely an omnibus PR (but you lagged the federal omnibus). 

Can you add some more details in the PR description on what this does, e.g.,
* Adds REST API 
* Moves points to database objects
* etc
* etc

I pulled, built, and tested locally. All worked.

We need to merge and release `alfalfa-client` with https://github.com/NREL/alfalfa-client/pull/47 and https://github.com/NREL/alfalfa-client/pull/45 before we publish this formerly. I can look at https://github.com/NREL/alfalfa-client/pull/45 if you want... it appears to be failing.

**Comment by TShapinsky** [comment]
> Looks great! This is definitely an omnibus PR (but you lagged the federal omnibus).
> 
> Can you add some more details in the PR description on what this does, e.g.,
> 
>     * Adds REST API
> 
>     * Moves points to database objects
> 
>     * etc
> 
>     * etc
> 
> 
> I pulled, built, and tested locally. All worked.
> 
> We need to merge and release `alfalfa-client` with [NREL/alfalfa-client#47](https://github.com/NREL/alfalfa-client/pull/47) and [NREL/alfalfa-client#45](https://github.com/NREL/alfalfa-client/pull/45) before we publish this formerly. I can look at [NREL/alfalfa-client#45](https://github.com/NREL/alfalfa-client/pull/45) if you want... it appears to be failing.

Yeah, I can add more details, there were more in one of the PRs which was merged into this

File: alfalfa_worker/jobs/modelica/step_run.py [hunk 5]
```diff
@@ -2,23 +2,22 @@
...
-        end_datetime = str(datetime(historian_year, 1, 1, 0, 0, 0) + timedelta(seconds=float(end_datetime)))
         super().__init__(run_id, realtime, timescale, external_clock, start_datetime, end_datetime)
         self.logger.info(f"{start_datetime}, {end_datetime}")
+        sim_year = self.start_datetime.year
+        self.sim_start_time = (self.start_datetime - datetime(sim_year, 1, 1)) / timedelta(seconds=1)
+        self.sim_end_time = (self.end_datetime - datetime(sim_year, 1, 1)) / timedelta(seconds=1)
 
-        self.current_datetime = datetime(historian_year, 1, 1, 0, 0, 0) + timedelta(seconds=self.sim_start_time)
-        print("current datetime at start of simulation: %", self.current_datetime)
+        print("current datetime at start of simulation: %", self.start_datetime)
```
[the remark this answers is not in the document]
Comment (TShapinsky) [reply alfalfa_worker/jobs/modelica/step_run.py:20]: should probably move it to a logger at the very least

**Review by nllong — APPROVED** [review]
I think this is ready to go!

**Pull request merged**